functions/sendgrid/index.js

```diff
@@ -13,101 +13,345 @@
+      var filename = uuid.v4() + '.json';
+      var file = storage.bucket(bucketName).file(filename);
+
+      console.log('Saving events to ' + filename + ' in bucket ' + bucketName);
+
+      return file.save(json, function (err) {
+        if (err) {
+          console.error(err);
+          return res.status(500).end();
+        }
+        console.log('JSON written to ' + filename);
+        return res.status(200).end();
+      });
+    }
+
+    return res.status(200).end();
+  } catch (err) {
+    console.error(err);
+    return res.status(err.code || 500).send(err.message);
+  }
+};
+// [END webhook]
+
+// [START getTable]
+/**
+ * Helper method to get a handle on a BigQuery table. Automatically creates the
+ * dataset and table if necessary.
+ *
+ * @param {Function} callback Callback function.
+ */
+function getTable (callback) {
+  var dataset = bigquery.dataset(config.DATASET);
+  return dataset.get({
+    autoCreate: true
+  }, function (err, dataset) {
+    if (err) {
+      return callback(err);
+    }
+    var table = dataset.table(config.TABLE);
+    return table.get({
+      autoCreate: true
+    }, function (err, table) {
+      if (err) {
+        return callback(err);
+      }
+      return callback(null, table);
+    });
+  });
+}
+// [END getTable]
+
+// [START load]
+/**
+ * Cloud Function triggered by Cloud Storage when a file is uploaded.
+ *
+ * @param {Object} context Cloud Function context.
+ * @param {Function} context.success Success callback.
+ * @param {Function} context.failure Failure callback.
+ * @param {Object} data Request data, in this case an object provided by Cloud Storage.
+ * @param {string} data.bucket Name of the Cloud Storage bucket.
+ * @param {string} data.name Name of the file.
+ * @param {string} [data.timeDeleted] Time the file was deleted if this is a deletion event.
+ * @see https://cloud.google.com/storage/docs/json_api/v1/objects#resource
+ */
+exports.sendgridLoad = function sendgridLoad (context, data) {
+  try {
+    if (data.hasOwnProperty('timeDeleted')) {
+      // This was a deletion event, we don't want to process this
+      return context.done();
+    }
+
+    if (!data.bucket) {
+      throw new Error('Bucket not provided. Make sure you have a ' +
+        '"bucket" property in your request');
+    }
+    if (!data.name) {
+      throw new Error('Filename not provided. Make sure you have a ' +
+        '"name" property in your request');
+    }
+
+    return async.waterfall([
+      // Get a handle on the table
+      function (callback) {
+        getTable(callback);
+      },
+      // Start the load job
+      function (table, callback) {
+        console.log('Starting job for ' + data.name);
+
+        var file = storage.bucket(data.bucket).file(data.name);
+        var metadata = {
+          autodetect: true,
+          sourceFormat: 'NEWLINE_DELIMITED_JSON'
+        };
+        table.import(file, metadata, callback);
+      },
+      // Poll the job for completion
```

Comment:
I'd like to see a comment here explaining that if they don't care about the result they can skip this.